### PR TITLE
Add dnsmasq as extraContainer to loki-gateway

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/giantswarm/issues/28813

### CODE REVIEW

File: helm/loki/values.yaml [hunk 1]
```diff
@@ -90,7 +90,18 @@ loki:
   enabled: true
 
   gateway:
+    extraContainers:
+      - name: dnsmasq
+        image: "janeczku/go-dnsmasq:release-1.0.7"
```
Comment (QuentinBisson) [inline helm/loki/values.yaml:95]: I think we will have to retag that image but otherwise looks good

**Review by QuentinBisson — APPROVED** [review]

File: helm/loki/values.yaml (hunk 1 above)
Comment (marieroque) [reply helm/loki/values.yaml]: => image retag and repo `go-dnsmasq` created on `quay` and `aliyun`

**Comment by marieroque** [comment]
WAIT FOR MERGE:
Asked opinion to @giantswarm/team-cabbage about the fix and potential side effects

**Comment by QuentinBisson** [comment]
Maybe ask on slack ?

**Comment by mcharriere** [comment]
hi there. I don't see any problem with this; but I'm a bit skeptical about the end result.

After reading the gist linked in the issue, I have 2 things to clarify:

> The issue manifests after the coredns pods backing the kube-dns service are restarted

NGINX doesn't talk directly to `coredns`; as we run `k8s-dns-local-cache` on each node. It's even more unlikely that the local cache gets restarted (without the entire node going down)

> Kubernetes uses DNAT to route/re-write packets destined from a service's ClusterIP to one of the backing pod's IP, which results in an entry being created in the conntrack table.

with Cilium this is a bit different. Cilium relies on `socket-LB` and doesn't keep a conntrack table the same way that the traditional k8s network stack does. 


but hey! if it works for you, go ahead :)

**Pull request merged**